extensions/BugModal/lib/WebService.pm

```diff
@@ -62,6 +81,26 @@ sub rest_resources {
     ]
 }
 
+sub products {
+    my $user = Bugzilla->user;
+    my @products = @{ $user->get_enterable_products };
+
+    @products = map { { name => $_->name } } @products;
+    return {
+        products     => \@products,
+    };
+}
+
+sub components {
+    my ($self, $params) = @_;
+    my $product = Bugzilla::Product->check({ name => $params->{product_name}, cache => 1 });
+    
+    my @components = map { { name => $_->name } } @{ $product->components };
```

Comment:
I had done that, but to keep uniformity with the product code and to avoid confusion, I wrote it this way. For your reference atoll had already reviewed with this similar query: https://github.com/mozilla-bteam/bmo/pull/103#pullrequestreview-44132271